stats/counter32.go

```diff
@@ -0,0 +1,36 @@
+package stats
+
+import (
+	"sync/atomic"
+	"time"
+)
+
+type Counter32 struct {
+	val uint32
+}
+
+func NewCounter32(name string) *Counter32 {
+	return registry.add(name, &Counter32{}).(*Counter32)
+}
+
+func (c *Counter32) SetUint32(val uint32) {
+	atomic.StoreUint32(&c.val, val)
+}
+
+func (c *Counter32) Inc() {
+	atomic.AddUint32(&c.val, 1)
+}
+
+func (c *Counter32) Add(val int) {
+	c.AddUint32(uint32(val))
+}
+
+func (c *Counter32) AddUint32(val uint32) {
+	atomic.AddUint32(&c.val, val)
```

Comment:
What happens if this `uint32` is full? Do we want it to just wrap around to `0` assuming that this will only happen if the user should have chosen a `uint64` instead?